---
title: "DDR Characterization"
author: "Ali Movasati"
date: "7/25/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r libraries, echo=FALSE, out.width = '100%', message = FALSE, warning=FALSE}

library(DT)
library(readxl)
library(dplyr)
library(stringr)


`%notin%` <- Negate(`%in%`)
```

```{r data, echo=FALSE, out.width = '100%', message = FALSE, warning=FALSE}


genes <- read_xlsx("/home/ali313/Documents/studies/master/umc-project/hpc/cuppen/projects/P0025_PCAWG_HMF/drivers/analysis/dna-rep-ann/external-files/43018_2020_50_MOESM3_ESM.xlsx", sheet = 7, skip = 1)
genes <- genes[,-5]

diplotype_final <- read.csv(file = "/home/ali313/Documents/studies/master/umc-project/hpc/cuppen/projects/P0025_PCAWG_HMF/drivers/analysis/dna-rep-ann/all-diplotypes-metadata-merged.txt.gz",
                            sep = "\t", header = T, stringsAsFactors = F)

metadata_included_ann <- read.csv(file = "/home/ali313/Documents/studies/master/umc-project/hpc/cuppen/projects/P0025_PCAWG_HMF/drivers/analysis/dna-rep-ann/r-objects/metadata-included-ddrd-annotated.txt",
            sep = "\t", header = T, stringsAsFactors = F)

metadata_included_ann$tmb <- rowSums(metadata_included_ann[,22:23])

```






## Introduction


In this report I'll go through details of mthodology and criteria used for determining DDR deficiency in cancer samples from HMF and PCAWG. I have also included preliminary results towards the end. 


## Overview


DDR pathways and genes that are studied come from the following paper: <https://doi.org/10.1038/s43018-020-0050-6>

Pleasance et al. determined deficiency of 12 DDR pathway examining somatic sequence of 181 genes involved in those pathways, whether as core or accessory genes. Some genes were involved in more than one pathway. Out of 181 genes, 4 genes whose entries were absent in grch37.ensembl were dropped ("BABAM2", "FAAP100", "FAAP20", "FAAP24"). These genes were only involved as accessory components. Therefore, in this study we will use 177 genes in total. A table of genes and their pathway(s) can be found below:

N.B. Genes that are accessory in some pathways and core in others have more than one row.

```{r ddr overview, echo=FALSE, message = FALSE, warning=FALSE}

# Remove genes that are not annotated
pruned_genes <- genes[genes$Gene_somatic %notin% c("BABAM2", "FAAP100", "FAAP20", "FAAP24"),]

pruned_genes %<>% group_by(Type, Gene_somatic) %>% summarise(Type = Type, Pathway = paste(Pathway, collapse = "|"), Pathway_abb = paste(Pathway_abb, collapse = "|")) %>% distinct() %>% relocate(Gene_somatic, .before = Type)

datatable(pruned_genes[order(pruned_genes$Gene_somatic),], options = list(
  searching = TRUE,
  pageLength = 10,
  lengthMenu = c(5, 10, 15, 20)
), caption = 'Table 1: DDR pathways and genes')

```


| For all these genes their biallelic status was determined by examining copy number information and simple mutations. Simple mutations were annotated using SnpEff and CLinVar databases. An overview of the scoring can be found below:



```{r clinvar-scoring, echo=FALSE, message = FALSE, warning=FALSE}

clinvar_scoring <- read.csv(file = "/home/ali313/Documents/studies/master/umc-project/hpc/cuppen/projects/P0025_PCAWG_HMF/drivers/analysis/dna-rep-ann/external-files/clinvar-scoring.txt", header = T, sep = "\t", stringsAsFactors = F)

datatable(clinvar_scoring, options = list(
  searching = TRUE,
  pageLength = 10,
  lengthMenu = c(5, 10, 15, 20)
), caption = 'Table 2: ClinVar Scoring')


```



```{r snp-scoring, echo=FALSE, message = FALSE, warning=FALSE}
snpeff_scoring <- read.csv(file = "/home/ali313/Documents/studies/master/umc-project/hpc/cuppen/projects/P0025_PCAWG_HMF/drivers/analysis/dna-rep-ann/external-files/snpeff-scoring.txt", header = T, sep = "\t", stringsAsFactors = F)

datatable(snpeff_scoring, options = list(
  searching = TRUE,
  rownames = FALSE,
  pageLength = 10,
  lengthMenu = c(5, 10, 15, 20)
), caption = 'Table 3: SnpEff Scoring')



```

***

| <b>selection criteria at the gene level.</b> Having the status of both alleles of each gene, we can decide whether we would like to consider a monoallelic hit as dysfunctionality criteria or a biallelic hit. If we consider a score of 3 or more for each allele to be dysfunctional, then will get 4986 genes that have biallelic dysfunctionality vs. 180,821 genes that have monoallelic dysfunctionality. If we consider a score of 4 or higher those number will be 3,006 and 176,116, respectively. Looking at the number of deficient samples in other similar studies, including the one we are using as reference, it makes more sense to use the biallelic hits to determine deficiency. Regarding the threshold score, as we are dealing with somatic events, I would continue with a score of 3 or higher.


- <i> overview of biallelic status when considering monoallelic hit (>=3) </i>:

```{r 2, echo=FALSE, out.width = '100%', message = FALSE, warning=FALSE}



table(diplotype_final$biall_status[(diplotype_final$a1.max_score >= 3 | diplotype_final$a2.max_score >= 3) & !is.na(diplotype_final$core_pathway)])


```


- <i> overview of biallelic status when considering beallelic hit (>=3) </i>:

```{r 22, echo=FALSE, out.width = '80%', message = FALSE, warning=FALSE}


table(diplotype_final$biall_status[(diplotype_final$a1.max_score >= 3 & diplotype_final$a2.max_score >= 3) & !is.na(diplotype_final$core_pathway)])


```




***


| <b>Moving on to selection criteria at the sample level.</b> at this stage, I have only considered core genes (n=112) of each pathway. In every sample and for each pathway, we count the number of genes that are likely disfunctional (having a score of 3 or more in both alleles). A dysfunctional gene that is a core component of more than one pathway will be counted towards each pathway.


| If a sample has a count of zero for all pathways (i.e. none of the core genes are determined as dysfunctional), the sample is annotated as "no deficiency". If in a sample two pathways have the same number of dysfunctional core genes (no absolute maximum), the sample is annotated as "no consensus". Finally, if one of the pathways has more dysfunctional core genes than others, the sample is annotated to be deficient for that pathway. 

- <i> Overview of DDR deficiency

N.B. SSA pathway is co-deficient in 54 four cases but in none of them it has the absolute maximum. That's because all 4 core genes in SSA are also core genes in other pathways and it's unlikely for SSA to be selected.

Overall the number of samples are lower comparing to Pleasance et al. work. This could partly be explained by the fact that our cohort (n=7049) insludes both primary (n=2310) and metastatic (n=4739) cancers while their cohort only consists of metastatic tumors (n=570).


```{r 3, echo=FALSE, message = FALSE, warning=FALSE}


a <- table(metadata_included_ann$dna_repair_deficiency[metadata_included_ann$is_metastatic])
b <- table(metadata_included_ann$dna_repair_deficiency[!metadata_included_ann$is_metastatic])
b["MMEJ"] <- 0
b <- b[c(1:5,13, 6:12)]
summ_def <- data.frame(row.names = names(a), Metastatic_count = as.vector(a), Primary_count = as.vector(b))
summ_def$Total_count <- rowSums(summ_def)

summ_def <- summ_def[c(11,10,1:9,12,13),]

summ_def$Metastatic_proportion <- (summ_def$Metastatic_count/nrow(metadata_included_ann[metadata_included_ann$is_metastatic,]))*100
summ_def$Primary_proportion <- (summ_def$Primary_count/nrow(metadata_included_ann[!metadata_included_ann$is_metastatic,]))*100
summ_def$Total_proportion <- (summ_def$Total_count/nrow(metadata_included_ann))*100

ff <- metadata_included_ann[metadata_included_ann$is_metastatic,] %>% group_by(dna_repair_deficiency) %>% summarize(Mean = mean(tmb, na.rm = T))
gg <- metadata_included_ann[metadata_included_ann$is_metastatic,] %>% group_by(dna_repair_deficiency) %>% summarize(Median = median(tmb, na.rm = T))

hh <- metadata_included_ann[!metadata_included_ann$is_metastatic,] %>% group_by(dna_repair_deficiency) %>% summarize(Mean = mean(tmb, na.rm = T))
jj <- metadata_included_ann[!metadata_included_ann$is_metastatic,] %>% group_by(dna_repair_deficiency) %>% summarize(Median = median(tmb, na.rm = T))

kk <- metadata_included_ann %>% group_by(dna_repair_deficiency) %>% summarize(Mean = mean(tmb, na.rm = T))
ll <- metadata_included_ann %>% group_by(dna_repair_deficiency) %>% summarize(Median = median(tmb, na.rm = T))

summ_def$Metastatic_Mean <- ff$Mean
summ_def$Metastatic_Median <- gg$Median

summ_def$Primary_Mean <- c(hh$Mean[1:5], 0, hh$Mean[6:12])
summ_def$Primary_Median <- c(jj$Median[1:5], 0, jj$Median[6:12])

summ_def$Total_Mean <- kk$Mean
summ_def$Total_Median <- ll$Median

summ_def <- summ_def[,c(1,4,7,8,2,5,9,10,3,6,11,12)]

datatable(summ_def, options = list(
  searching = TRUE,
  rownames = FALSE,
  pageLength = 15,
  lengthMenu = c(5, 10, 15, 20)
), caption = 'Table 3: SnpEff Scoring')


```



### Compare TMB of DDR-deficient groups

```{r 4, echo=FALSE, out.width = '100%', message = FALSE, warning=FALSE}


img1_path <- "/home/ali313/Documents/studies/master/umc-project/hpc/cuppen/projects/P0025_PCAWG_HMF/drivers/analysis/dna-rep-ann/explore/figs/pdf/jpg/ddrd-tmb-boxplot-1.jpg"
knitr::include_graphics(img1_path)

```
